client/src/components/Modals/SignUpModal.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from 'react'
+import gql from 'graphql-tag'
+import PropTypes from 'prop-types'
+import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap'
+import { Mutation } from 'react-apollo'
+import '../NavBar/NavBar.css'
+
+const newUserMutation = gql`
+  mutation NewUser($TeacherName: String!, $TeacherEmail: String!, $TeacherPW: String!) {
+    createTeacher(TeacherName: $TeacherName, TeacherEmail: $TeacherEmail, TeacherPW: $TeacherPW) {
+      teacher {
+        TeacherName
+      }
+      jwtString
+    }
+  }`
+
+class SignUpModal extends Component {
+  constructor () {
+    super()
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      confirmPassword: '',
+      passwordError: '',
+      signUpError: ''
+    }
+  }
+
+  handleInputChange = event => {
+    this.setState({ [event.target.name]: event.target.value })
+  }
+
+  render () {
+    const { name, email, password, confirmPassword } = this.state
+    return (
+      <Modal className='signup_login_modal' isOpen={this.props.signUpModal} toggle={this.props.toggleSignUp}>
+        <ModalHeader className='signup_login_modal_header'>
+          <span>Sign up for free to create study sets</span>
+        </ModalHeader>
+        <Mutation mutation={newUserMutation}>
+          {(createNewUser, { loading }) => (
+            <form onSubmit={event => {
+              event.preventDefault()
+              if (this.state.password !== this.state.confirmPassword) {
+                this.setState({ passwordError: 'Passwords do not match.' })
+              } else {
+                const createdUser = createNewUser({ variables: { TeacherName: name, TeacherEmail: email, TeacherPW: password } })
+                createdUser.then(() => {
+                  this.setState({
+                    name: '',
+                    email: '',
+                    password: '',
+                    confirmPassword: '',
+                    passwordError: false
+                  })
+                  this.props.toggleSignUp()
+                  this.props.toggleLogIn()
```

Comment:
Forgot to mention this in the previous review, why toggle the login component after signup? Wouldn't it make for a better user experience to simply save the JWT and take them right into the app? That's how it was constructed before, I'm curious why this was changed...